From 6fcb3ff0038008f0befdcb8a24e5d79064f255f5 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sat, 19 Mar 2022 16:53:59 -0500
Subject: [PATCH] mlir: set CMAKE_INCLUDE_CURRENT_DIR to fix out-of-tree builds

Required include directories from build tree
were previously added in
but need to be added ourselves now.

This isn't needed when building MLIR with LLVM_ENABLE_PROJECTS
as the build within LLVM's setting of this option.

Various other LLVM projects do this as well.
---
 mlir/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mlir/CMakeLists.txt b/mlir/CMakeLists.txt
index 1c1a281bf4fe..00e5606d92df 100644
--- a/mlir/CMakeLists.txt
+++ b/mlir/CMakeLists.txt
@@ -139,6 +139,8 @@ if(MLIR_ENABLE_BINDINGS_PYTHON)
   mlir_configure_python_dev_packages()
 endif()
 
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
 include_directories( "include")
 include_directories( ${MLIR_INCLUDE_DIR})
 
-- 
2.35.1

